[wled] Cannot invoke "java.util.List.iterator()" because "this.state.jsonResponse.palettes" is null

Since upgrading to OH 5.2.0 WLED segment things fail to beome online. Until OH 5.1.4 it worked without problems. I also tested a plain new installation wirhout success.

## Expected Behavior
As in versions earlier than OH 5.2.0, the individual segments should switch to the "online" status.

## Current Behavior
Once the bridge has been successfully connected to the WLED device (online), the segments appear in the inbox.
However, the individual segments do not go online, and the error "Cannot invoke 'java.util.List.iterator()' because 'this.state.jsonResponse.palettes' is null" is displayed.

<img width="712" height="155" alt="Image" src="https://github.com/user-attachments/assets/1745c9ae-1c76-453e-994f-70adbbb792de" />

<img width="545" height="72" alt="Image" src="https://github.com/user-attachments/assets/10f7c0a2-d401-47bc-810e-9301be00f9ea" />

Log:
```
[INFO] [openhab.event.ThingStatusInfoChangedEvent] - Thing 'wled:segment:test_wled:test_wled-0' changed from UNINITIALIZED (DISABLED) to INITIALIZING
[INFO] [openhab.event.ThingStatusInfoChangedEvent] - Thing 'wled:segment:test_wled:test_wled-0' changed from INITIALIZING to ONLINE
[ERROR] [org.openhab.core.internal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.wled.internal.handlers.WLedSegmentHandler@3eb21530': Cannot invoke "java.util.List.iterator()" because "this.state.jsonResponse.palettes" is null
[INFO] [openhab.event.ThingStatusInfoChangedEvent] - Thing 'wled:segment:test_wled:test_wled-0' changed from ONLINE to UNINITIALIZED (HANDLER_INITIALIZING_ERROR): Cannot invoke "java.util.List.iterator()" because "this.state.jsonResponse.palettes" is null
[ERROR] [org.openhab.core.thing.internal.ThingManagerImpl] - Exception occurred while initializing handler of thing 'wled:segment:test_wled:test_wled-0': Cannot invoke "java.util.List.iterator()" because "this.state.jsonResponse.palettes" is null
```

## Possible Solution
No workaround was found yet.

## Steps to Reproduce (for Bugs)
1. Install [WLED binding](https://www.openhab.org/addons/bindings/wled/)
2. Add a `wled:json` thing
3. Go to the inbox
4. Add a `wled:segment` that was found in the inbox as a thing
5. Observe the thing status of the segment and have a look in the log files.

## Context
Probably a regression since the latest changes of this binding introduced in OH 5.2.0?

## Your Environment
openHAB 5.2.0 Release Build
Main UI Commit [36efc977](https://github.com/openhab/openhab-webui/commit/36efc977)
Tested with openhabian and windows
WLED v16.0.1

--- a/bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WledState.java
+++ b/bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WledState.java
@@ -38,8 +38,8 @@ public class WledState {
     public PresetState[] presetState = new PresetState[1];
 
     public class JsonResponse {
-        public List<String> effects = new ArrayList<>();
-        public List<String> palettes = new ArrayList<>();
+        public @Nullable List<String> effects = new ArrayList<>();
+        public @Nullable List<String> palettes = new ArrayList<>();
     }
 
     public void unpackJsonObjects() {

--- a/bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/api/WledApiV084.java
+++ b/bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/api/WledApiV084.java
@@ -225,7 +225,7 @@ public class WledApiV084 implements WledApi {
             }
         } else {
             logger.debug(
-                    "Effects in JSON response are empty (either bridge isn't initialized properly (yet) or WLED firmware does not return them)");
+                    "Effects in JSON response are missing (either bridge isn't initialized properly (yet) or WLED firmware does not return them)");
         }
         return fxOptions;
     }
@@ -244,7 +244,7 @@ public class WledApiV084 implements WledApi {
             }
         } else {
             logger.debug(
-                    "Palettes in JSON response are empty (either bridge isn't initialized properly (yet) or WLED firmware does not return them)");
+                    "Palettes in JSON response are missing (either bridge isn't initialized properly (yet) or WLED firmware does not return them)");
         }
         return palleteOptions;
     }

--- a/bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/api/WledApiV084.java
+++ b/bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/api/WledApiV084.java
@@ -215,11 +215,17 @@ public class WledApiV084 implements WledApi {
     public List<StateOption> getUpdatedFxList() {
         List<StateOption> fxOptions = new ArrayList<>();
         int counter = 0;
-        for (String value : state.jsonResponse.effects) {
-            fxOptions.add(new StateOption(Integer.toString(counter++), value));
-        }
-        if (handler.config.sortEffects) {
-            fxOptions.sort(Comparator.comparing(o -> "0".equals(o.getValue()) ? "" : o.getLabel()));
+        List<String> responseEffects = state.jsonResponse.effects;
+        if (responseEffects != null) {
+            for (String value : responseEffects) {
+                fxOptions.add(new StateOption(Integer.toString(counter++), value));
+            }
+            if (handler.config.sortEffects) {
+                fxOptions.sort(Comparator.comparing(o -> "0".equals(o.getValue()) ? "" : o.getLabel()));
+            }
+        } else {
+            logger.debug(
+                    "Effects in JSON response are empty (either bridge isn't initialized properly (yet) or WLED firmware does not return them)");
         }
         return fxOptions;
     }
@@ -228,11 +234,17 @@ public class WledApiV084 implements WledApi {
     public List<StateOption> getUpdatedPaletteList() {
         List<StateOption> palleteOptions = new ArrayList<>();
         int counter = 0;
-        for (String value : state.jsonResponse.palettes) {
-            palleteOptions.add(new StateOption(Integer.toString(counter++), value));
-        }
-        if (handler.config.sortPalettes) {
-            palleteOptions.sort(Comparator.comparing(o -> "0".equals(o.getValue()) ? "" : o.getLabel()));
+        List<String> responsePalettes = state.jsonResponse.palettes;
+        if (responsePalettes != null) {
+            for (String value : responsePalettes) {
+                palleteOptions.add(new StateOption(Integer.toString(counter++), value));
+            }
+            if (handler.config.sortPalettes) {
+                palleteOptions.sort(Comparator.comparing(o -> "0".equals(o.getValue()) ? "" : o.getLabel()));
+            }
+        } else {
+            logger.debug(
+                    "Palettes in JSON response are empty (either bridge isn't initialized properly (yet) or WLED firmware does not return them)");
         }
         return palleteOptions;
     }

--- a/bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java
+++ b/bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java
@@ -285,7 +285,11 @@ public class WLedSegmentHandler extends BaseThingHandler {
             WledApi localAPI = localBridgeHandler.api;
             if (localAPI != null) {
                 updateStatus(ThingStatus.ONLINE);
-                updateStateDescriptionProviders();
+                if (bridge.getStatus() == ThingStatus.ONLINE) {
+                    // only update the providers once the bridge is ready, otherwise this will be done once it goes
+                    // online via bridgeStatusChanged()
+                    updateStateDescriptionProviders();
+                }
                 if (!localBridgeHandler.hasWhite) {
                     logger.debug("WLED is not setup to use RGBW, so removing un-needed white channels");
                     removeWhiteChannels();

